tests/FileModelTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace OCA\Analysis_App\Tests;
+
+use OCA\Analysis_app\FileModel;
+use OCP\Files\Folder;
+use OCP\Files\Node;
+use Test\TestCase;
+
+class FileModelTest extends TestCase {
+    /** @var \PHPUnit_Framework_MockObject_MockObject | Folder $currentFolderMock */
+    private $currentFolderMock;
+    /** @var FileModel $fileModelObject */
+    private $fileModelObject;
+    /** @var Node | \PHPUnit_Framework_MockObject_MockObject */
+    private $nodeMock1;
+    /** @var Node | \PHPUnit_Framework_MockObject_MockObject */
+    private $nodeMock2;
+
+    protected function setUp() {
+        parent::setUp();
+        $this->nodeMock1 = $this->getMockBuilder(Node::class)
+            ->getMock();
+        $this->nodeMock2 = $this->getMockBuilder(Node::class)
+            ->getMock();
+        $this->currentFolderMock = $this->getMockBuilder(Folder::class)
+            ->getMock();
+        $this->fileModelObject = new FileModel($this->currentFolderMock);
+    }
+
+    public function testAnalyze() {
+        $this->nodeMock1->expects($this->any())
+            ->method('getName')->will($this->returnValue('nodemock1.pdf'));
+        $this->currentFolderMock->expects($this->any())
+            ->method('search')
+            ->will($this->returnValue([$this->nodeMock1, $this->nodeMock2]));
+        $this->fileModelObject->analyze($this->currentFolderMock);
+    }
+
+    public function testGetAnalysisReport() {
+        $this->nodeMock1->expects($this->any())->method('getName')
+            ->will($this->returnValue('nodeMock1.pdf'));
+        $this->nodeMock2->expects($this->any())->method('getSize')
+            ->will($this->returnValue(20));
+        $this->currentFolderMock
+            ->expects($this->any())
+            ->method('search')
+            ->will($this->returnValue([$this->nodeMock1, $this->nodeMock2]));

```

Comment:
You can assert equal to the report result array by creating expected result array

Reply:
This is not quite possible because I couldn't be able to configure a mockobject by giving it two different methods, so they weren't able to return 2 different values (i.e. size and name at the same time for each mock)

Reply:
AFAIK, your assumption is wrong, it should work pretty straightforward. Example is given below;
````
$object = $this->getMock('object', array('set_properties','get_events'));

$object
  ->expects($this->once())
  ->method('get_events')
  ->will($this->returnValue(array()));
$object
  ->expects($this->once())
  ->method('set_properties')
  ->with($this->equalTo(array()));